Allow filtering of dataset by years

mzagaja opened the pull request:
This commit allows the user to filter the dataset by values available in the years column in the user interface. This also addresses #1 by removing the isMunicipal property of the model since it was merely checking to see if the table had the word “Municipal” in it to then sort by municipality.

Resolves mapc/datacommon#65

mzagaja committed:
Allow filtering of dataset by years

This commit allows the user to filter the dataset by values available in the years column in the user interface. This also addresses #1 by removing the isMunicipal property of the model since it was merely checking to see if the table had the word “Municipal” in it to then sort by municipality.

Resolves mapc/datacommon#65

mzagaja committed:
Enable ES6 Sourcemaps

Using instructions at https://github.com/babel/ember-cli-babel#enabling-source-maps to enable easier debugging

mzagaja committed:
Display SQL API errors

* Refactor years available to dynamically use the correct years column name in datasets.js
* Refactor template to display Ember errors if AJAX request to SQL API returns a permission error due to the dataset being private

sukima commented on app/controllers/datasets.js:
```diff
@@ -17,7 +17,7 @@ export default Ember.Controller.extend({
   download_link: Ember.computed('model', 'model.years_available.@each.selected', function() {
     let yearsSelected = this.get('model.years_available') || [];
-    if (!yearsSelected.errors) {
+    if (Ember.compare(Ember.Error, yearsSelected) !== 0) {
```
What is this? I've never compared an error to something before? Did you mean `yearsSelected instanceof Ember.Error`?

sukima commented on app/controllers/datasets.js:
```diff
@@ -76,10 +76,13 @@ export default Ember.Controller.extend({
-    let emberModelObject = this.get('model');
     return this.get('model.raw_data.rows').filter(function(row) {
-      return years_available.includes(row.acs_year);
+      return years_available.includes(eval(year_column));
```
The coding police have been called. `eval` to lookup a value? `eval` that is not sandboxed? There is something fundamentally odd going on here.

mzagaja committed:
Evict Eval

Do not use eval because it is unsafe per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval#Do_not_ever_use_eval

mzagaja committed:
Refactor Error Code

Refactor error detection code to be more clear.

mzagaja committed:
Fix Ember Error Comparison + CORS

* Resolve checking the instanceOf an Ember.Error class by using uppercase “Error” because Error is an Ember class and thus needs to be uppercase.

* Resolve a CORS issue by pinging the HTTPS endpoint for datacommon metadata instead of the HTTP endpoint.

ericyoungberg commented on app/templates/datasets.hbs:
```diff
@@ -107,10 +98,24 @@
+            </h3>
+          {{else}}
+            {{ui-table model=(slice min max selected_rows) fields=model.raw_data.fields metadata=formattedMetadata}}
+            }
```
This is a rogue bracket that renders at the bottom of the table.

ericyoungberg commented on ember-cli-build.js:
```diff
@@ -4,7 +4,9 @@ const EmberApp = require('ember-cli/lib/broccoli/ember-app');
-    // Add options here
+    babel: {
+        sourceMaps: 'inline'
+      }
```
We don't want inline source maps for production builds. Use `EmberApp.env()` for deciding whether this value should be set or not.

ericyoungberg reviewed (changes requested):
The first part of the issue is complete but there are two more things to do:

1. Only the latest year should be selected by default. The other years shouldn't be selected on initial render.
2. Except for the csv, the date range for the file links don't update as you (de)select years.

mzagaja committed:
Evict the Rogue Bracket

I have no idea why this keeps re-appearing despite the fact I believe I have deleted it several times.

mzagaja committed:
Set Sourcemaps Based on Environment

mzagaja committed:
Start with only first year selected in a data set

Resolve MAPC/datacommon#65 by only selecting the first year.

mzagaja committed:
Reverse Sort the Array

Reverse sort the array so that we always select the most recent year instead of the least recent year.

mzagaja committed:
Minor refactor

Use let instead of var

mzagaja committed:
Let user select no years

This lets users select a dataset with no years instead of throwing a console error when they try to de-select the last year.
